### ISSUE TITLE
UserSession. completeOAuth2 doesn't work when popup is true

### ISSUE THREAD

**Issue by pfcstyle** [Esri/arcgis-rest-js#711]
The oauth2 popup window didn't close and callback function didn't be called.

arcgis-rest-auth version: 2.13.0
Chrome version: Version 83.0.4103.97 (Official Build) (64-bit)

I noticed this in `UserSession.ts`:
```
    if (
          popup &&
          win.opener &&
          win.opener.parent &&
          win.opener.parent[`__ESRI_REST_AUTH_HANDLER_${clientId}`]
        ) {
          const handlerFn =
            win.opener.parent[`__ESRI_REST_AUTH_HANDLER_${clientId}`];
          if (handlerFn) {
            handlerFn(
              error ? JSON.stringify(error) : undefined,
              JSON.stringify(oauthInfo)
            );
          }
          win.close();
          return undefined;
        }
```

But I found the `__ESRI_REST_AUTH_HANDLER_${clientId}` was exists in win.opener only, so it can't get the handleFn.

**Comment by pfcstyle** [comment]
So I submit the pr  to  get handlerFn from win.opener if it can't get the callback from win.opener.parent. I think it can resolve this issue and won't result in other issues.

```
    if (
          popup &&
          win.opener &&
          win.opener.parent &&
          (win.opener.parent[`__ESRI_REST_AUTH_HANDLER_${clientId}`] || win.opener[`__ESRI_REST_AUTH_HANDLER_${clientId}`])
        ) {
          const handlerFn =
            win.opener.parent[`__ESRI_REST_AUTH_HANDLER_${clientId}`] || win.opener[`__ESRI_REST_AUTH_HANDLER_${clientId}`];
          if (handlerFn) {
            handlerFn(
              error ? JSON.stringify(error) : undefined,
              JSON.stringify(oauthInfo)
            );
          }
          win.close();
          return undefined;
        }
```

https://github.com/Esri/arcgis-rest-js/pull/712

**Comment by dbouwman** [comment]
Thanks @pfcstyle - I was literally about to open an issue about this *exact* issue!

To date this has not been an issue but we've started doing some dev't work using [Storybook](https://storybook.js.org/) and that loads the components into iframes. When we started using auth in that scenario, we ran into this issue where window.parent (the page hosting the iframe) did not have the handler, but window.opener (the page loaded into the iframe) did.

Thanks for the PR - we'll review it asap and move to get it merged.
